content/blog/automation-api-as-platform/index.md

```diff
@@ -0,0 +1,275 @@
+            res.status(404).send(`stack "${stackName}" does not exist`);
+        } else if (e instanceof ConcurrentUpdateError) {
+            res.status(409).send(`stack "${stackName}" already has update in progress`)
+        } else {
+            res.status(500).send(e);
+        }
+    }
+};
+const ensurePlugins = async () => {
+    const ws = await LocalWorkspace.create({});
+    await ws.installPlugin("aws", "v3.2.1");
+};
+
+// install necessary plugins once upon boot
+ensurePlugins();
+
+// configure express
+const app = express();
+app.use(express.json());
+```
+
+Creating routes in Express to the callbacks is straightforward and requires mapping the route to the appropriate callback.
+
+```typescript
+// setup our RESTful routes for our Site resource
+app.post("/sites", createHandler);
+app.get("/sites", listHandler);
+app.get("/sites/:id", getHandler);
+app.put("/sites/:id", updateHandler);
+app.delete("/sites/:id", deleteHandler);
+
+// start our http server
+app.listen(1337, () => console.info("server running on :1337"));
+```
+
+## Trying it out
+
+The complete example is available on [GitHub](https://github.com/pulumi/automation-api-examples/tree/main/nodejs/pulumiOverHttp-ts). along with other examples in both Nodejs and go. To run the TypeScript example, run:
+
+```bash
+$ yarn install
+$ yarn start
+```
+
+Let's test it by sending requests to the server on `http://localhost:1337`.
+
+{{% notes type="info" %}}
+If you use [Postman](https://www.postman.com/), you can import a [Postman Collection](https://gist.github.com/pulumipus/56d1ee83f295971e2a26a8091880c482) to send requests to the server.
+{{% /notes %}}
+
+First let's create our first website which returns the website path (`id`) and the URL of the static website.
+
+```bash
+$ curl --header "Content-Type: application/json"   --request POST   --data '{"id":"hello","content":"hello world\n"}'   http://localhost:1337/sites
+$ {"id":"hello","url":"s3-website-bucket-db5dae8.s3-website-us-west-2.amazonaws.com"}
+```
+
+We can view the content then update it, then view it again.
+
+```bash
+$ curl s3-website-bucket-db5dae8.s3-website-us-west-2.amazonaws.com
+$ curl --header "Content-Type: application/json"   --request PUT   --data '{"id":"hello","content":"hello updated world!\n"}'   http://localhost:1337/sites/hello
+$ curl s3-website-bucket-db5dae8.s3-website-us-west-2.amazonaws.com
+```
+
+Once we're done, we can delete the website.
+
+```bash
+$ curl --header "Content-Type: application/json"   --request DELETE http://localhost:1337/sites/hello
+```
+
+## Pulling it all together
+
+So what can you do with this? We used a static website in this example so that results are immediately available. However, this application does basic [CRUD](https://en.wikipedia.org/wiki/Create,_read,_update_and_delete) operations, which are the basis of many applications. For example, an application could send photos to a bucket or send data to a database application.
```

Comment:
> For example, an application could send photos to a bucket or send data to a database application.

I think this would be a good point to link back to the title "Build Self-Service Cloud Infrastructure with Automation API".

Automation API enables building our own custom cloud platforms. This example has shown us how to take our customized unit of infrastructure and expose it to our broader team via a REST interface that we define and control.  This interface is familiar to developers, and feels more comfortable than using IaC directly. Today, we expose a static website, but tomorrow it could be a Kubernetes application platform with all of our networking and security best practices.

Reply:
added